pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationState.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pulsar.broker.authentication;
+
+import java.io.IOException;
+import javax.naming.AuthenticationException;
+
+/**
+ * Interface for authentication state.
+ *
+ * It is basically holding the the authentication state.
+ * It tell broker whether the authentication is completed or not,
+ * if completed, what is the AuthRole is.
+ */
+public interface AuthenticationState {
+    AuthenticationDataSource getAuthData();
+
+    /**
+     * After the authentication between client and broker completed,
+     * get authentication role represent for the client.
+     *
+     */
+    String getAuthRole() throws AuthenticationException;
```

Comment:
What kind of values we are expecting here?

Reply:
It is a role token String that used in Pulsar authr framework.
[This link](http://pulsar.apache.org/docs/en/security-overview) mentioned this role token a little